svelte/src/scripts/draw/filter-node.ts

```diff
@@ -0,0 +1,36 @@
+import type {GraphDataNode} from '$types';
+
+// Will filter node recursively
+export default function filterNodeAndPopulateFilteredID(
+	nodes: GraphDataNode[],
+	filteredNodeNames: string[],
+): GraphDataNode[] {
+	if (nodes.length < 1) return [];
+	const filteredNodes = nodes.filter(node => {
+		if (filteredNodeNames.includes(node.id)) {
+			node.members.forEach(member => {
+				filteredNodeNames.push(member.id);
+				populateFilterId(member.members, filteredNodeNames);
+			});
+			return false;
+		}
+		return true;
+	});
+
+	for (let i = 0; i < filteredNodes.length; i++) {
+		filteredNodes[i].members = filteredNodes[i].originalMembers;
```

Comment:
this means that when a node is filtered and then removed, they will always be uncollapsed, right?

Reply:
I'm not really sure what you mean by always be uncollapsed